tests/api-watch/test_diff_modtpe_decl.ml

```diff
@@ -0,0 +1,75 @@
+open Api_watch
+open Test_helpers
+
+let%expect_test "Module_type with value changes" =
+  let reference =
+    compile_interface
+      {|
+    module type M = sig
+      val g : int -> string
+    end
+  |}
+  in
+  let current = compile_interface {|
+    module type M 
+  |} in
+  let result = Diff.interface ~module_name:"Main" ~reference ~current in
+  Format.printf "%a" pp_diff_option result;
+  [%expect
+    {| Some (Module Main: {Modified (Supported [ Module_type M: {Modified (Unsupported)}])}) |}]
+
+let%expect_test "Module_type with module change" =
+  let reference =
+    compile_interface
+      {|
+    module type M = sig
+      val b : int list -> int
+    end
+    module P : sig val x: float end
+  |}
+  in
+  let current =
+    compile_interface
+      {|
+    module type M = sig
+      val b : float list -> float
+    end
+    module type N = sig val x: int end
```

Comment:
Did you add this module type to test module type addition? It's best to keep test cases minimal and focused on a specific feature, code path or regression.

Module type addition should have its own test case by comparing an empty interface with one that has a module type.